examples/all_reduce.py

```diff
@@ -0,0 +1,171 @@
+from __future__ import annotations
+
+import os
+
+import torch
+import torch.distributed as dist
+import torch.distributed._symmetric_memory as symm_mem
+import triton
+import triton.language as tl
+
+import helion
+import helion.language as hl
+
+
+# Symmemtric Memory Helpers
+@triton.jit
+def triton_copy(
+    inp: tl.int64,  # pyright: ignore[reportInvalidTypeForm]
+    out: tl.tensor,
+    SIZE: tl.constexpr,
+) -> None:
+    tl.static_assert(out.dtype.is_ptr())
+    inp = inp.to(tl.pointer_type(out.dtype.element_ty))  # pyright: ignore[reportAttributeAccessIssue]
+    addrs = tl.load(inp + tl.arange(0, SIZE))
+    tl.store(out + tl.arange(0, SIZE), addrs)
+
```

Comment:
Workaround before symmetric memory handler natively supports getting `dev_ptrs` as a tensor, instead of a pointer to the array.